semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala

```diff
@@ -11,55 +12,77 @@ import scala.util.control.NonFatal
 import org.langmeta.internal.io._
 import org.langmeta.io._
 import java.util.concurrent.atomic.AtomicBoolean
+import scala.collection.GenSeq
 
 class Main(settings: Settings, reporter: Reporter) {
   def process(): Option[Classpath] = {
     val success = new AtomicBoolean(true)
-    val outs = {
-      settings.classpath.shallow.par.flatMap { in =>
-        if (in.isDirectory) {
-          val out = AbsolutePath(Files.createTempDirectory("semanticdb"))
-          val res = convertClasspathEntry(in, out)
-          success.compareAndSet(true, res)
-          List(out)
-        } else if (in.isFile) {
-          val cacheEntry = {
-            val base = settings.cacheDir
-            val checksum = Checksum(in)
-            base.resolve(in.toFile.getName.stripSuffix(".jar") + "-" + checksum + ".jar")
-          }
-          if (cacheEntry.toFile.exists) {
-            List(cacheEntry)
-          } else {
-            PlatformFileIO.withJarFileSystem(cacheEntry) { out =>
-              val res = convertClasspathEntry(in, out)
-              success.compareAndSet(true, res)
-              List(cacheEntry)
-            }
-          }
+
+    val classpath: GenSeq[AbsolutePath] =
+      if (settings.par) settings.classpath.shallow.par
+      else settings.classpath.shallow
+
+    val buffer = new ConcurrentLinkedQueue[AbsolutePath]()
+
+    classpath.foreach { entry =>
+      val processedEntry: Option[AbsolutePath] = if (entry.isDirectory) {
+        val out = AbsolutePath(Files.createTempDirectory("semanticdb"))
+        val res = convertClasspathEntry(entry, out)
+        success.compareAndSet(true, res)
+        Some(out)
```

Comment:
`buffer.add(out)`? This will allow to remove the need to declare `processedEntry` as well as the pattern matching bit on lines 51-54.